dask_saturn/core.py

```diff
@@ -23,6 +45,30 @@ def __init__(self, cluster_url=None, *args, **kwargs):
         self.loop = None
         self.periodic_callbacks = {}
 
+    @classmethod
+    def reset(
+            cls,
+            worker_size=None,
+            scheduler_size=None,
+            nprocs=None,
+            nthreads=None,
+            scheduler_service_wait_timeout=DEFAULT_WAIT_TIMEOUT_SECONDS
+        ):
+        """Destroys an existing Dask cluster attached to the Jupyter Notebook or
+        Custom Deployment, and recreates it with the given configuration"""
+        print(f"Resetting cluster.")
+        url = urljoin(BASE_URL, "api/dask_clusters/reset")
+        cluster_config = {
+            "worker_size": worker_size,
+            "scheduler_size": scheduler_size,
+            "nprocs": nprocs,
+            "nthreads": nthreads,
+        }
+        response = requests.post(url, data=json.dumps(cluster_config), headers=HEADERS)
+        if not response.ok:
+            raise ValueError(response.reason)
+        return cls()
```

Comment:
don't you need to get the `cluster_url` from the response and pass it in here?

Reply:
nope. Cluster hasn't actually started yet, it is just starting. Passing in cluster_url would tell it that the cluster is ready, and it would then immediately fail on trying to hit the status route. By passing in nothing, it goes into the _start loop where it polls for the existing cluster to reach the ready state

Reply:
oh ok, got it